### PR TITLE
Deploy new  min_max_scaler function

### PR DESCRIPTION
**Pull request by julienawonga**
This function performs min-max scaling on an array. It takes an array as input and returns the scaled values between 0 and 1

### CODE REVIEW

File: bigfunctions/min_max_scaler.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      |    1    |   0.0   |
+      |    4    |   0.75  |
+      |    3    |   0.5   |
+      |    2    |   0.25  |
+      |    5    |   1.0   |
+      +---------+---------+
+code: |
+  const min = Math.min(...arr);
+  const max = Math.max(...arr);
+  return arr.map(x => (x - min) / (max - min));
```
Comment (AxelThevenot) [inline bigfunctions/min_max_scaler.yaml:32]: EOF

**Comment by unytics** [comment]
Thanks @julienawonga and @AxelThevenot for this great function. 
I merge it and I will deploy it.

**Comment by julienawonga** [comment]
Thanks @AxelThevenot  , @unytics  !

**Comment by unytics** [comment]
**The function is deployed and works congrats 👏 @julienawonga and @AxelThevenot **
Here's [the link](https://unytics.io/bigfunctions/bigfunctions/min_max_scaler/) in the documentation.

On the other hand, when I tried to deploy the function, I got some errors that I needed to fix. Then once deployed when I tested it, I got some other errors.
Please next time, try to deploy the function on your side, test it, and solve errors before creating the pull request.

(I should definitely add automatic testing in a test environment in github actions to prevent that).

If you have difficulties in solving errors, don't hesitate to have a chat with me [on slack](https://join.slack.com/t/unytics/shared_invite/zt-1gbv491mu-cs03EJbQ1fsHdQMcFN7E1Q)

Errors were linked to the fact that bigquery do not support js functions with `any type`  as input or output. 
image


I converted it into a sql function which works but also, is much more efficient than js on BigQuery.
[Here](https://github.com/unytics/bigfunctions/blob/main/bigfunctions/min_max_scaler.yaml#L22) is the new code.

**Comment by AxelThevenot** [comment]
My bad I made a review with the comment but I forgot to click on the request change button so you thought it was ready as only the EOF comment has appeared

**Comment by julienawonga** [comment]
Thank you @unytics  .  
I try this first, and to avoid passing min, max as args, I just write Js code. Thanks a Lot!
--
WITH Data AS (
  SELECT val
  FROM UNNEST([2, 3, 5, 8]) AS val
),
MinMaxValues AS (
  SELECT
    MIN(val) AS min_val,
    MAX(val) AS max_val
  FROM Data
)

-- 
CREATE TEMP FUNCTION MinMaxScaler(x FLOAT64, min_val FLOAT64, max_val FLOAT64) 
RETURNS FLOAT64
AS (
  (x - min_val) / (max_val - min_val)
);

--
SELECT
  d.val,
  MinMaxScaler(d.val, mm.min_val, mm.max_val) AS Scaled
FROM
  Data d,
  MinMaxValues mm;

**Comment by julienawonga** [comment]
image

**Comment by julienawonga** [comment]
image

**Comment by AxelThevenot** [comment]
@julienawonga 
You can try to create the standard_scaler (if not yet exist) 


```sql
select big functions.eu.standard_scaler(arr)
```

This time opening first an issue then deploying your function locally using the `bigfun` 

It has to work before opening the pull request 🙏
Do not hesitate if you have questions or have some issues !

**Comment by julienawonga** [comment]
Thank you ! I'll work on it.

**Pull request merged**